pkg/kubelet/kubelet_node_status.go

```diff
@@ -946,7 +890,7 @@ func (kl *Kubelet) defaultNodeStatusFuncs() []func(*v1.Node) error {
 		nodestatus.NodeAddress(kl.nodeIP, kl.nodeIPValidator, kl.hostname, kl.externalCloudProvider, kl.cloud, nodeAddressesFunc),
 		withoutError(kl.setNodeStatusInfo),
 		nodestatus.OutOfDiskCondition(kl.clock.Now, kl.recordNodeStatusEvent),
-		withoutError(kl.setNodeMemoryPressureCondition),
+		nodestatus.MemoryPressureCondition(kl.clock.Now, kl.evictionManager.IsUnderMemoryPressure, kl.recordNodeStatusEvent),
```

Comment:
rather than defining each pressure as a func, can we define an interface `NodeResourcePressureProvider` (or whatever you want to name it), which includes all the `IsUnderXPressure` functions, and pass `kl.evictionManager` as the implementation of this interface.  I prefer passing interfaces to functions whenever possible.